<?xml version="1.0"?>

<robot name="transmitter" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <xacro:arg name="id" default="0" />
	<xacro:arg name="frame_id" default="world" />
	
	<!-- Base Link -->
  <link name="transmitter">
    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
      <geometry>
        <box size="0.3 0.1 0.04"/>
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 0.0" rpy="0 0 0"/>
      <geometry>
        <box size="0.3 0.1 0.04"/>
      </geometry>
      <material name="orange"/>
    </visual>

    <inertial>
      <origin xyz="0 0 1" rpy="0 0 0"/>
      <mass value="1"/>
      <inertia
        ixx="1.0" ixy="0.0" ixz="0.0"
        iyy="1.0" iyz="0.0"
        izz="1.0"/>
    </inertial>
  </link>

  <material name="orange">
    <color rgba="${255/255} ${108/255} ${10/255} 1.0"/>
  </material>
 
	<gazebo reference="transmitter">
    <material>Gazebo/Orange</material>
  </gazebo>		

	<gazebo>
		<plugin name="arva_transmitter" filename="libarva_sim_transmitter.so">
      <id>$(arg id)</id>
			<frame_id>$(arg frame_id)</frame_id>
		</plugin>
    <static>true</static>
  </gazebo>
</robot>
